language/vm/vm-genesis/src/lib.rs

```diff
@@ -531,9 +532,14 @@ pub fn encode_genesis_transaction_with_validator(
                 "Discovery set in emitted event does not match discovery set fed into genesis transaction",
             );
 
-            ChangeSet::new(txn_output.write_set().clone(), txn_output.events().to_vec())
+            ChangeSet::new(
+                0,
+                txn_output.write_set().clone(),
+                txn_output.events().to_vec(),
+            )
         }
     };
+    let hash = genesis_write_set.get_hash();
```

Comment:
In general, I'm not a big fan of changing the definition of functions unless it is used pervasively. It seems like adding the hash is a helper, but we can also just call hash() on the resulting blob. Perhaps that is a cleaner?